app/controllers/studies_controller.rb

```diff
@@ -445,7 +445,9 @@ def new_study_file
   # method to perform chunked uploading of data
   def do_upload
     upload = get_upload
-    filename = upload.original_filename
+    # remove spaces and + (encoded whitespace character) from filename since this converted client-side,
+    # but original_filename is unchanged in headers
+    filename = upload.original_filename.gsub(/[\s\+]/, '_')
```

Comment:
Does this mean that the names of downloaded files would have a "_" if the original file name had a space?

If so, names of downloaded files would not match those of uploaded files in that rare case.  But I can't think of any realistic concrete cases where that would be a issue -- so this seems OK.

Reply:
This really is only affecting the query when trying to figure out if this is a "new" upload, or the next chunk in an "existing" upload.  The filename conversion already happens client-side in the jQuery upload code, but since the `original_filename` header is unchanged and is being used for the query, that mismatch caused the query to return `nil`, which then throws an error when it tries to insert another `StudyFile` with the converted filename.

The end result is that _all_ uploaded files have spaces removed from their filenames, and that's what shows up in the download tab.  This has been the behavior for uploads in SCP since the beginning.